src/components/Explorer/SamplesByDataTypeTable.js

```
@@ -25,9 +24,16 @@ class SamplesByDataTypeTable extends Component {
             'Single-nucleus RNA-seq (snRNA-seq)': 'sn',
             'Single-cell RNA-seq (scRNA-seq)': 'sc',
             'Regional transcriptomics (LMD RNA-seq)': 'rt',
+            'Spatial metabolomics': 'sm',
+            'Spatial lipidomics': 'sl',
+            'Spatial N-Glycomics': 'sng'
         };
 
-        this.props.setSelectedConcept(dataLinkageMapping[dataType], this.props);
+        if (dataLinkageMapping[dataType]) {
+            this.props.setSelectedConcept(dataLinkageMapping[dataType], this.props);
+        } else {
+            throw new Error('Datatype not found', dataType)
```

What does the user see if we hit this error? Would this take them to the Oops page?

---

Currently, before this change, when a user clicks the link without an available dataLinkageMapping, the page would throw and return an error page (note, not an oops page).

With this change, the user is not redirected, instead they only view a link that does nothing when clicked. The error message is visible in the window's console output.

In addition to suppressing the error page, we could send the user to the oops page. This would be in the place of the user clicking a link and seeing nothing happen.

---

I would rather have the user go to the oops page than nothing happen. 
Ideally this would then also send an error to the service layer that we could capture and be notified about. However, I don't believe we have an endpoint currently to capture and log front-end errors for this application. So, at the very least, let's make sure they go to the oops page.

---

Sounds good to me, I'll update to navigate the user to the oops page.

I believe you're correct with the regard to the error capturing. We have handlers for networking errors, but I don't think we have anything for app-level errors. It might be worth writing a card to expand capture functionality to app-level, happy to write this up if it sounds good to you.